From eae76eb28d2d601f33677b890bf2b8fbf388ff04 Mon Sep 17 00:00:00 2001
From: Ryo Nakano <ryonakaknock3@gmail.com>
Date: Sat, 24 Aug 2024 18:05:01 +0900
Subject: [PATCH] AppearancePane: Remove access to Gala's appearance schema

---
 src/Panes/AppearancePane.vala | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/Panes/AppearancePane.vala b/src/Panes/AppearancePane.vala
index 6ccbf3d..10ece21 100644
--- a/src/Panes/AppearancePane.vala
+++ b/src/Panes/AppearancePane.vala
@@ -11,7 +11,7 @@ public class PantheonTweaks.Panes.AppearancePane : Categories.Pane {
     private Gee.HashMap<string, string> preset_button_layouts;
     private XSettings x_settings;
     private GLib.Settings interface_settings;
-    private GLib.Settings appearance_settings;
+    private GLib.Settings gnome_wm_settings;
 
     private Categories.Pane.ComboBoxText gtk_combobox;
     private Categories.Pane.ComboBoxText controls_combobox;
@@ -29,8 +29,7 @@ public class PantheonTweaks.Panes.AppearancePane : Categories.Pane {
         var sound_settings = new GLib.Settings ("org.gnome.desktop.sound");
         x_settings = new XSettings ();
         var gtk_settings = new GtkSettings ();
-        appearance_settings = new GLib.Settings ("org.pantheon.desktop.gala.appearance");
-        var gnome_wm_settings = new GLib.Settings ("org.gnome.desktop.wm.preferences");
+        gnome_wm_settings = new GLib.Settings ("org.gnome.desktop.wm.preferences");
 
         Pantheon.AccountsService? pantheon_act = null;
 
@@ -180,7 +179,6 @@ public class PantheonTweaks.Panes.AppearancePane : Categories.Pane {
 
         controls_combobox.changed.connect (() => {
             string new_layout = controls_combobox.active_id;
-            appearance_settings.set_string ("button-layout", new_layout);
             gnome_wm_settings.set_string ("button-layout", new_layout);
             x_settings.set_gnome_menu (gnome_menu.state, new_layout);
         });
@@ -201,7 +199,6 @@ public class PantheonTweaks.Panes.AppearancePane : Categories.Pane {
             }
 
             sound_settings.reset ("theme-name");
-            appearance_settings.reset ("button-layout");
             gnome_wm_settings.reset ("button-layout");
             x_settings.reset ();
 
@@ -212,7 +209,7 @@ public class PantheonTweaks.Panes.AppearancePane : Categories.Pane {
 
     private void init_data () {
         gtk_combobox.active_id = interface_settings.get_string ("gtk-theme");
-        controls_combobox.active_id = appearance_settings.get_string ("button-layout");
+        controls_combobox.active_id = gnome_wm_settings.get_string ("button-layout");
         gnome_menu.state = x_settings.has_gnome_menu ();
     }
 
-- 
2.43.0

